### ISSUE TITLE
Instance of killer being null when it should not be, in onMobDeath

### ISSUE THREAD

**Issue by Zenny89** [DarkstarProject/darkstar#1561]
I read issue #1538 and have repeated it numerous times on my SCH. Cast a helix on a worm, so that it takes 4-5 ticks to kill the worm. After the second to last tick the worm becomes unclaimed. On the last tick, the worm dies unclaimed, and I do not receive any experience points.

**Comment by TeoTwawki** [comment]
I would think the real issue here is that the mob came unclaimed before it should have.

**Comment by teschnei** [comment]
why should it not have? if you cast helix and then do nothing, why is it surprising that it eventually became unclaimed?

**Comment by TeoTwawki** [comment]
Because its still dott'ed and dotted mobs don't unclaimed normally? Is helix diff from other dots like that on retail? You tell me.

**Comment by teschnei** [comment]
it's not, but i don't think a dot keeps claim

**Comment by TeoTwawki** [comment]
It'll eventually declaim,  but just poison is usually enough to hold something for quite a while before it happens. I just assumed he's reporting it because it happened faster than normal.

**Comment by TeoTwawki** [comment]
Otherwise...unclaimed mob, not supposed to grant you exp anyway, and killer is correct to be null/nil.

**Comment by Zenny89** [comment]
Yes, it does seem to come unclaimed really fast. I haven't timed it, but 4-5 ticks of a helix doesn't seem like much time at all.

**Comment by bendangelo** [comment]
Does poison and other dots hold claim on retail? I don't remember.

**Comment by TeoTwawki** [comment]
I I don't know if its dot itself or just the enmity + a short grace period (unless you switch targets, then you lose it instantly thats another thing dsp doesn't have yet).

I could be wrong though. Best to get someone on retail to try a few diff duration dots while pinning and see if they always stay past the duration and past the enmity generated.
